polyfill/intersection-observer.js

```diff
@@ -618,7 +627,7 @@ IntersectionObserver.prototype._computeTargetAndRootIntersection =
  */
 IntersectionObserver.prototype._getRootRect = function() {
   var rootRect;
-  if (this.root) {
+  if (this.root && !isDoc(this.root)) {
```

Comment:
If we're doing it this way, then I think we need to change the code below:

```
var html = document.documentElement;
var body = document.body;
```

To something like:
```
const doc = isDoc(this.root) ? this.root : document;
var html = doc.documentElement;
var body = doc.body;
```

Reply:
definitely! similar misunderstanding of potentially using a child or parent's document for same-origin cases

Reply:
done